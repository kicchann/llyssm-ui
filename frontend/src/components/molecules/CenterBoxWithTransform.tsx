import { Box, styled } from '@mui/material';
import React from 'react';
import { TransformComponent, TransformWrapper } from 'react-zoom-pan-pinch';

interface CenterBoxWithTransformProps {
  children: React.ReactNode;
  wrapperProps?: object; // 必要に応じて外部からズームやパンの設定を上書きできるように
}

const StyledCenterBox = styled(Box)`
  position: relative;
  flex: 1;
  display: flex;
  justify-content: center;
  align-items: center;
  padding: 8;
`;

const defaultWrapperProps = {
  disabled: false,
  initialScale: 1,
  doubleClick: { mode: 'reset' as 'reset' },
  panning: { disabled: false, activationKeys: [], excluded: [] },
  pinch: { excluded: [], step: 5 },
  wheel: { activationKeys: [], excluded: [], smoothStep: 0.001, step: 0.2 },
};

export const CenterBoxWithTransform: React.FC<CenterBoxWithTransformProps> = ({
  children,
  wrapperProps = defaultWrapperProps, // デフォルトのwrapperPropsを用意
}) => {
  return (
    <StyledCenterBox>
      <TransformWrapper {...wrapperProps}>
        <TransformComponent>{children}</TransformComponent>
      </TransformWrapper>
    </StyledCenterBox>
  );
};
