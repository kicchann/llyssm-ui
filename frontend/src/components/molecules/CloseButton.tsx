import React from 'react';
import { GenericIconButton } from '../atoms/GenericIconButton';

interface CloseButtonProps {
  onClick: () => void;
}

export const CloseButton: React.FC<CloseButtonProps> = ({ onClick }) => {
  return (
    <GenericIconButton
      iconType="close"
      onClick={onClick}
      title="close"
      aria-label="close"
    />
  );
};
