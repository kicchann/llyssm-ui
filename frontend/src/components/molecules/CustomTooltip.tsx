import React from 'react';
import styled from 'styled-components';
import { MarkerData } from '../../types/map';

// スタイルの定義
// CustomTooltipはあとで静的なHTMLとして描画するため、styled-componentsを使ってスタイルを定義
// MUIは使わない
const StyledWrapper = styled.div`
  max-width: none;
  padding: 0;
  box-shadow: 0 0 0 2px white;
`;

const StyledImage = styled.img`
  width: 100%;
  border-radius: 4px 4px 0 0;
  padding: 30;
`;

const StyledContent = styled.article`
  margin: 1rem;
  text-align: justify;
`;

const StyledTitle = styled.h2`
  margin: 0;
`;

const StyledText = styled.p`
  margin: 0;
`;

interface CustomTooltipProps {
  markerData: MarkerData;
}

export const CustomTooltip: React.FC<CustomTooltipProps> = ({ markerData }) => {
  return (
    <>
      {/* Tooltip内容 */}
      <StyledWrapper className="custom-tooltip">
        <StyledImage src={markerData.thumbnailUrl} alt={markerData.name} />
        <StyledContent>
          <StyledTitle>{markerData.name}</StyledTitle>
          <StyledText>{markerData.description}</StyledText>
        </StyledContent>
      </StyledWrapper>
    </>
  );
};
