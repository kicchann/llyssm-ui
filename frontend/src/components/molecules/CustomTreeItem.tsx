import { styled } from '@mui/material';
import { TreeItem } from '@mui/x-tree-view/TreeItem';
import React from 'react';

interface CustomTreeItemProps {
  id: string;
  label: string;
  children?: React.ReactNode;
  onItemClick: (event: React.SyntheticEvent, itemId: string) => void;
  isSelected?: boolean;
}

const StyledTreeItem = styled(TreeItem)<{ isSelected?: boolean }>`
  & .MuiTreeItem-content {
    padding: 4px 0;
  }
  color: black;
  background-color: ${({ isSelected }) =>
    isSelected ? 'rgba(0, 0, 0, 0.1)' : 'transparent'};
`;

export const CustomTreeItem: React.FC<CustomTreeItemProps> = ({
  id,
  label,
  children,
  onItemClick,
  isSelected,
}) => {
  return (
    <StyledTreeItem
      isSelected={isSelected}
      key={id}
      itemId={id}
      label={label}
      onClick={(event) => onItemClick(event, id)}
    >
      {children}
    </StyledTreeItem>
  );
};
