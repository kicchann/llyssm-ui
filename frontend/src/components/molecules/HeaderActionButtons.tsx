import { Box, styled } from '@mui/material';
import React from 'react';

interface HeaderActionButtonsProps {
  children: React.ReactNode;
}

const ButtonContainer = styled(Box)`
  display: flex;
  gap: 8px; // ボタン間のスペースを統一
  justify-content: flex-end; // ボタンを右寄せ
`;

// 簡易なコンポーネントのため、テストは省略
export const HeaderActionButtons: React.FC<HeaderActionButtonsProps> = ({
  children,
}) => {
  return <ButtonContainer>{children}</ButtonContainer>;
};
