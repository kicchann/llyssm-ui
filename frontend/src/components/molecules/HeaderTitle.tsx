import { styled } from '@mui/material';
import React from 'react';
import { Label } from '../atoms/Label';

// レイアウトに関するスタイルは HeaderTitle 内で定義
const StyledHeaderTitle = styled('div')`
  flex-grow: 1;
`;

interface HeaderTitleProps {
  text: string;
}

// 簡易なコンポーネントのため、テストは省略
export const HeaderTitle: React.FC<HeaderTitleProps> = ({ text }) => {
  return (
    <StyledHeaderTitle>
      <Label text={text} color="white" variant="h6" />
    </StyledHeaderTitle>
  );
};
