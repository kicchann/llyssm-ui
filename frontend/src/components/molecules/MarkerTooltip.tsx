import React from 'react';
import styled from 'styled-components';
import { MarkerData } from '../../types/marker';

// スタイルの定義
// MarkerTooltipはあとで静的なHTMLとして描画するため、styled-componentsを使ってスタイルを定義
// MUIは使わない
const StyledWrapper = styled.div`
  max-width: none;
  padding: 0;
  box-shadow: 0 0 0 2px white;
`;

const StyledImage = styled.img`
  width: 100%;
  border-radius: 4px 4px 0 0;
  padding: 30;
`;

const StyledContent = styled.article`
  margin: 1rem;
  text-align: justify;
`;

const StyledTitle = styled.h2`
  margin: 0;
`;

const StyledText = styled.p`
  margin: 0;
`;

interface MarkerTooltipProps {
  markerData: MarkerData;
}

export const MarkerTooltip: React.FC<MarkerTooltipProps> = ({ markerData }) => {
  return (
    <>
      {/* Tooltip内容 */}
      <StyledWrapper className="marker-tooltip">
        <StyledImage src={markerData.thumbnailUrl} alt={markerData.name} />
        <StyledContent>
          <StyledTitle>{markerData.name}</StyledTitle>
          <StyledText>{markerData.description}</StyledText>
        </StyledContent>
      </StyledWrapper>
    </>
  );
};
