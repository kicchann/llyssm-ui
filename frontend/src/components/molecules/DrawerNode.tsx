import { Box, Drawer, styled } from '@mui/material';
import React, { useState } from 'react';
import { CloseButton } from './CloseButton';
import { ToggleDrawerButton } from './ToggleDrawerButton';

const DrawerBox = styled(Box)`
  width: 300px;
  padding: 16px;
  padding-top: 48px; /* 閉じるボタン分のスペースを確保 */
  position: relative;
`;

interface DrawerNodeProps {
  children: React.ReactNode;
}

export const DrawerNode: React.FC<DrawerNodeProps> = ({ children }) => {
  const [isOpen, setIsOpen] = useState(false);

  const toggleDrawer = (open: boolean) => () => {
    setIsOpen(open);
  };

  return (
    <div data-testid="drawer-node">
      {/* メニューを開くボタン */}
      <ToggleDrawerButton isOpen={isOpen} onToggle={toggleDrawer(true)} />

      {/* 右からスライドするメニュー */}
      <Drawer anchor="right" open={isOpen} onClose={toggleDrawer(false)}>
        <DrawerBox role="presentation">
          {/* 閉じるボタン */}
          <CloseButton onClick={toggleDrawer(false)} />
          {children}
        </DrawerBox>
      </Drawer>
    </div>
  );
};
