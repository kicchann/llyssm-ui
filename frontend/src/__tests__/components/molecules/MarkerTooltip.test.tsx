import { render } from '@testing-library/react';
import { MarkerTooltip } from '../../../components/molecules/MarkerTooltip';
import { MarkerData } from '../../../types/marker';

const mockMarkerData: MarkerData = {
  id: '1',
  sphereId: '1',
  name: 'Sample Marker',
  description: 'This is a sample marker description.',
  orientation: { yaw: 0, pitch: 0 },
  imageUrl: 'https://example.com/image.jpg',
  thumbnailUrl: 'https://example.com/image.jpg',
  markerType: 'image',
  createdAt: '2021-01-01T00:00:00.000Z',
};

describe('MarkerTooltip', () => {
  it('renders correctly with valid marker data', () => {
    const { container } = render(<MarkerTooltip markerData={mockMarkerData} />);
    expect(container).toMatchSnapshot();
  });
});
