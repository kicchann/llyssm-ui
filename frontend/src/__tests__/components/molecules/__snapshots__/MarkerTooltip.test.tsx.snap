// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MarkerTooltip renders correctly with valid marker data 1`] = `
<div>
  <div
    class="sc-blHHSb heJFIm marker-tooltip"
  >
    <img
      alt="Sample Marker"
      class="sc-gtLWhw kqDegA"
      src="https://example.com/image.jpg"
    />
    <article
      class="sc-egkSDF ikupuP"
    >
      <h2
        class="sc-fAUdSK eMUwlG"
      >
        Sample Marker
      </h2>
      <p
        class="sc-dntaoT bzATOg"
      >
        This is a sample marker description.
      </p>
    </article>
  </div>
</div>
`;
